articles/compliance/gdpr/features-aiding-compliance/user-consent.md

```diff
@@ -0,0 +1,353 @@
+
+---
+
+:::panel What else do I have to do?
+- Determine how you want to track consent. We recommend putting in not just the date but the correct version of terms and conditions, specifying the version, and including an array for those who will withdraw, provide and withdraw consent again.
+- Choose where you want to store consent, in Auth0's database or elsewhere
+:::
+
+## Withdraw consent
+
+The user should have the option to withdraw consent using your app. This option should be easily accessible, and clearly distinguishable. Once the users decides to withdraw their consent, you should take action. First, you have to decide how you will handle withdrawal of consent: will you delete the users or flag them as deleted?
+
+### Delete user
+
+To delete a user, use the [Delete a user endpoint](/api/management/v2#!/Users/delete_users_by_id). 
+
+```har
+{
+    "method": "DELETE",
+    "url": "https://${account.namespace}/api/v2/users/USER_ID",
+    "httpVersion": "HTTP/1.1",
+    "headers": [{
+        "name": "Authorization",
+        "value": "Bearer YOUR_MGMT_API_ACCESS_TOKEN"
+    }]
+}
+```
+
+The response body for this endpoint is empty, so if you want to confirm that the user was successfully deleted try to [retrieve the user using their email](/users/search#users-by-email). If the endpoint returns an error, then your call to delete the user was successful.
+
+### Flag user as deleted
+
+If you don't want to completely delete the user, flag their profile as deleted at the `app_metadata` (endpoint: [Update a user](/api/management/v2#!/Users/patch_users_by_id)). Then, add some code that will make the authentication process to fail for any user with their profile flagged as such. This way you can keep a record of deleted users in case you need to refer to this information in the future.
+
+#### Step 1: Flag the profile
+
+To flag a user as deleted use the [app_metadata](/metadata). In this example we will add a metadata property `deleted` which when set to `true` means that the user should be treated as deleted.
+
+To update a user's metadata use the [Update a user endpoint](/api/management/v2#!/Users/patch_users_by_id).
+
+```har
+{
+    "method": "PATCH",
+    "url": "https://${account.namespace}/api/v2/users/USER_ID",
+    "httpVersion": "HTTP/1.1",
+    "headers": [{
+        "name": "Authorization",
+        "value": "Bearer YOUR_MGMT_API_ACCESS_TOKEN"
+    },
+    {
+        "name": "content-type",
+        "value": "application/json"
+    }],
+    "postData" : {
+        "mimeType": "application/json",
+        "text": "{\"app_metadata\":{\"deleted\":true}}"
+    }
+}
+```
+
+#### Step 2: Disable login for flagged users
+
+In this step we need to disable login for users flagged as deleted. To do so we will add a JavaScript snippet that will run as part of the authentication pipeline. We call these snippets [rules](/rules).
```

Comment:
Next, we need to disable login for users flagged as deleted. To do so, we will add a [rule](/rules), or a JavaScript snippet, that will run as part of the authentication pipeline